src/Ilicop.Web/GwpProcessor.cs

```diff
@@ -21,29 +23,96 @@ public class GwpProcessor : IProcessor
     private readonly DirectoryInfo configDir;
     private readonly ILogger<GwpProcessor> logger;
     private readonly GwpProcessorOptions gwpProcessorOptions;
+    private readonly IlitoolsExecutor ilitoolsExecutor;
 
-    public GwpProcessor(IOptions<GwpProcessorOptions> gwpProcessorOptions, IFileProvider fileProvider, ILogger<GwpProcessor> logger)
+    public GwpProcessor(
+        IOptions<GwpProcessorOptions> gwpProcessorOptions,
+        IFileProvider fileProvider,
+        IlitoolsExecutor ilitoolsExecutor,
+        ILogger<GwpProcessor> logger)
     {
         this.fileProvider = fileProvider ?? throw new ArgumentNullException(nameof(fileProvider));
         this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
         this.gwpProcessorOptions = gwpProcessorOptions?.Value ?? throw new ArgumentNullException(nameof(gwpProcessorOptions));
+        this.ilitoolsExecutor = ilitoolsExecutor ?? throw new ArgumentNullException(nameof(ilitoolsExecutor));
 
         if (this.gwpProcessorOptions.ConfigDir != null)
             this.configDir = new DirectoryInfo(this.gwpProcessorOptions.ConfigDir);
     }
 
     /// <inheritdoc />
-    public void Run(Guid jobId, Profile profile)
+    public async Task Run(Guid jobId, string transferFile, Profile profile, CancellationToken cancellationToken)
     {
         if (configDir == null || !Directory.Exists(Path.Combine(configDir.FullName, profile.Id)))
         {
             logger.LogInformation("No configuration directory found for profile <{ProfileId}>. Skipping GWP processing for job <{JobId}>.", profile.Id, jobId);
             return;
         }
 
+        await CreateGpkg(jobId, transferFile, profile, cancellationToken);
         CreateZip(jobId, profile);
     }
 
+    private async Task CreateGpkg(Guid jobId, string transferFile, Profile profile, CancellationToken cancellationToken)
+    {
+        var dataGpkgFilePath = Path.Combine(configDir.FullName, profile.Id, gwpProcessorOptions.DataGpkgFileName);
+
+        if (!File.Exists(dataGpkgFilePath))
+        {
+            logger.LogWarning("No data GeoPackage file found at <{GpkgFilePath}> for profile <{ProfileId}>. Skipping GWP GeoPackage creation for job <{JobId}>.", dataGpkgFilePath, profile.Id, jobId);
+            return;
+        }
+
+        fileProvider.Initialize(jobId);
+        var destGpkgFilePath = Path.Combine(fileProvider.HomeDirectory.FullName, gwpProcessorOptions.DataGpkgFileName);
+
+        using (var destGpkgFileStream = fileProvider.CreateFile(destGpkgFilePath))
+        using (var sourceGpkgFileStream = File.OpenRead(dataGpkgFilePath))
+        {
+            sourceGpkgFileStream.CopyTo(destGpkgFileStream);
+        }
```

Comment:
PP: Diesen Teil in `TryCopyTemplateGpkg `auslagern und innerhalb von `Run` die schritte explizit aufrufen. So liest sich das ganze im Run sehr einfach
```
if (TryPrepareTemplateGpkg){
  ImportData()
  ImportErrors()
  CopyQgisServerFile()
  CreateZip()
}
```